controller/src/main/java/org/jboss/as/controller/transform/description/AttributeTransformationDescriptionBuilderImpl.java

```diff
@@ -29,74 +30,72 @@ abstract class AttributeTransformationDescriptionBuilderImpl<T extends BaseAttri
 
     @Override
-    public ResourceTransformationDescriptionBuilder end() {
-        return builder;
+    public Stability getStability() {
+        return this.builder.getStability();
     }
 
     @Override
-    public T setDiscard(DiscardAttributeChecker discardChecker, AttributeDefinition...discardedAttributes) {
-        AttributeDefinition[] useDefs = discardedAttributes;
-        for (AttributeDefinition attribute : useDefs) {
-            String attrName = getAttributeName(attribute);
-            registry.setDiscardedAttribute(discardChecker, attrName);
-        }
-        return thisBuilder();
+    public ResourceTransformationDescriptionBuilder end() {
+        return builder;
     }
 
     @Override
     public T setDiscard(DiscardAttributeChecker discardChecker, Collection<AttributeDefinition> discardedAttributes) {
-        for (AttributeDefinition attribute : discardedAttributes) {
-            String attrName = getAttributeName(attribute);
-            registry.setDiscardedAttribute(discardChecker, attrName);
+        for (AttributeDefinition discardedAttribute : discardedAttributes) {
+            if (this.enables(discardedAttribute)) {
+                this.registry.setDiscardedAttribute(discardChecker, discardedAttribute.getName());
+            }
         }
-        return thisBuilder();
+        return this.thisBuilder();
     }
 
     @Override
     public T setDiscard(DiscardAttributeChecker discardChecker, String... discardedAttributes) {
-        String[] useDefs = discardedAttributes;
-        for (String attrName : useDefs) {
-            registry.setDiscardedAttribute(discardChecker, attrName);
+        for (String discardedAttribute : discardedAttributes) {
+            this.registry.setDiscardedAttribute(discardChecker, discardedAttribute);
         }
         return thisBuilder();
     }
 
     @Override
-    public T addRejectCheck(final RejectAttributeChecker checker, final AttributeDefinition...rejectedAttributes){
-        for (AttributeDefinition attribute : rejectedAttributes) {
-            String attrName = getAttributeName(attribute);
-            registry.addAttributeCheck(attrName, checker);
+    public T addRejectCheck(final RejectAttributeChecker checker, final AttributeDefinition... rejectedAttributes){
+        for (AttributeDefinition rejectedAttribute : rejectedAttributes) {
+            if (this.enables(rejectedAttribute)) {
+                this.registry.addAttributeCheck(rejectedAttribute.getName(), checker);
+            }
         }
-        return thisBuilder();
+        return this.thisBuilder();
     }
 
     @Override
     public T addRejectCheck(RejectAttributeChecker rejectChecker, String... rejectedAttributes) {
-        for (String attribute : rejectedAttributes) {
-            registry.addAttributeCheck(attribute, rejectChecker);
+        for (String rejectedAttribute : rejectedAttributes) {
+            this.registry.addAttributeCheck(rejectedAttribute, rejectChecker);
         }
```

Comment:
Should this method be deprecated? Just reading the diff it seems like it's unable to apply the desired 'enables' check so it's just assuming. Which is ok temporarily but perhaps should be deprecated.

It seems like the javadoc in the interface should explain the behavior re the 'enables' call for all the methods.

Reply:
#6589 adds the deprecation.

Reply:
Thanks for that - I'll rebase this branch on that change next week.